### ISSUE TITLE
Remove gogo + migrate to protobufv2

### ISSUE THREAD

**Issue by ericvn** [istio/istio#26010]
**Bug description**
```
istioctl profile dump demo
2020-07-31T14:30:50.531650Z	info	proto: tag has too few fields: "-"
! MeshConfig.disablePolicyChecks is deprecated. Mixer is deprecated and will be removed from Istio with the 1.8 release. Please consult our docs on the replacement.
```

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

No ` proto: tag has too few fields: "-"` message

**Steps to reproduce the bug**

Run `istioctl profile dump demo`

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

1.7.0-alpha.2

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

MacOS

**Comment by howardjohn** [comment]
@esnible or @ostromart may know the magic incantation to make this go away

**Comment by sdake** [comment]
This problem has been present since Istio 1.4. There was a recent PR to optimize the loading that removed this error, although it has been reverted because it doesn't handle some specific scenarios of the operator.

I'd treat this as a warning for now and ignore. A different option is to just delete the error message, create a P0 bug, and mark it staleproof.

Cheers,
-steve

**Comment by esnible** [comment]
I believe this is a flaw in a Kubernetes library.  See my comment in the closed issue https://github.com/istio/istio/issues/19735#issuecomment-589305545 .  I am not sure why that issue is closed.

There are two ways to fix the library.  If an APIMachinery maintainer suggested which direction to take I would take it.  I feel silly issuing two completely different PRs for the same problem.

I tried to patch over this with Istio by sending output to a buffer instead of stderr.  This worked great and the fix was small!  Unfortunately it failed tests because of a race condition ... redirecting error output while tests run in parallel suppressed more than `proto: tag has too few fields: "-"`.  We could pursue that approach further, but it seems silly when the fix could be just a few lines in another library.

**Comment by JimmyCYJ** [comment]
I am going to mark this as P0 as it affects installation

**Comment by howardjohn** [comment]
I would treat this lower priority, it does not impact users and is not a
common command

**Comment by JimmyCYJ** [comment]
@howardjohn Thanks, I put a P1 to this issue, please feel free to change.

**Comment by ldemailly** [comment]
my two cents would be not to capture/hide the message but get a fix of the root cause

**Comment by esnible** [comment]
> my two cents would be not to capture/hide the message but get a fix of the root cause

@ldemailly A key Kubernetes library says https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go#L34 :
```
type Time struct {
	time.Time `protobuf:"-"`
}
```

I suspect gogo/protobuf expected `""` not `"-"`, for this situation. 
  See https://github.com/gogo/protobuf/blob/master/proto/properties.go#L368 and https://github.com/gogo/protobuf/blob/master/proto/properties.go#L207

We could also avoid this by switching 100% to golang/protobuf.  Istio uses "gogo/protobuf" 180 times and "golang/protobuf" 164 times.

**Comment by howardjohn** [comment]
> We could also avoid this by switching 100% to golang/protobuf. Istio uses
"gogo/protobuf" 180 times and "golang/protobuf" 164 times.

Does this solve the problem? K8s uses gogo

**Comment by sdake** [comment]
@ericvn this seems to have the most context of any of the issues related to gogo. As a result, I am reopening this issue, renaming it, and linking the work in this area to transition from gogo.

**Comment by howardjohn** [comment]
duplicate of https://github.com/istio/istio/issues/26751, lets track in one place
